server/jetstream_test.go

```diff
@@ -21517,3 +21517,156 @@ func TestJetStreamLimitsToInterestPolicy(t *testing.T) {
 	require_Equal(t, info.State.FirstSeq, 11)
 	require_Equal(t, info.State.Msgs, 10)
 }
+
+func TestJetStreamConsumerDefaultsFromStream(t *testing.T) {
```

Comment:
I would add tests for Clustered mode too, to ensure we have the same behavior in clustered setup.